### DISCUSSION TITLE
Writing Streams

### DISCUSSION THREAD

**Discussion by spunkyvt** [dokan-dev/dokan-dotnet#342] [Q&A]
WIth the original Mirror from Dokan I can use PS and c# to write streams.  I cannot get the sample dotnet one to write streams.  The error it give me is The given path's format is not supported.  I have tried several ways

The powershell I used was 

$$file="F:\stream.txt"
$streamName= "myStream"

$textToAdd = "Hello World"

$textToAdd | Add-Content -Path "$file':$streamName"

This works with the original mirror, but cannot get it to work with the doakanNetMirror.

**Comment by Olof-Lagerkvist** [comment]
How do you implement the create file and write methods in your .NET class? There are several limitations for opening and writing alternate data streams in .NET applications, particularly in old .NET Framework but there are some things to watch out for in modern .NET versions as well.

**Comment by spunkyvt** [comment]
I left them as they were from the sample app.  I am compiling this in .Net 4.7.2

**Reply by Olof-Lagerkvist** [reply]
In .NET Framework you cannot access alternate data streams using .NET library functions at all. You need to create your own wrapper functions and call native `CreateFile` for that.

Or, switch to target a .NET Core or .NET 5+ runtime.
